src/Microsoft.Extensions.Logging.EventSource/LoggingEventSource.cs

```diff
@@ -208,29 +191,168 @@ protected override void OnEventCommand(EventCommandEventArgs command)
+                    }
+
+                    if (parts.Length == 2)
+                    {
+                        if (!TryParseLevel(defaultLevel, parts[1], out level))
+                        {
+                            continue;
+                        }
+                    }
+
+                    rules.Add(new LoggerFilterRule(typeof(EventSourceLoggerProvider).FullName, loggerName, level, null));
+                }
+            }
+
+            return rules.ToArray();
+        }
+
+        /// <summary>
+        /// Parses the level specification (which should look like :N where n is a  number 0 (Trace)
+        /// through 5 (Critical).   It can also be an empty string (which means 1 (Debug) and ';' marks
+        /// the end of the specifcation This specification should start at spec[curPos]
+        /// It returns the value in 'ret' and returns true if successful.  If false is returned ret is left unchanged.
+        /// </summary>
+        [NonEvent]
+        private static bool TryParseLevel(LogLevel defaultLevel, string levelString, out LogLevel ret)
+        {
+            ret = defaultLevel;
+
+            if (levelString.Length == 0)
+            {
+                // No :Num spec means Debug
+                ret = defaultLevel;
+                return true;
+            }
+
+            int level;
+            switch (levelString)
+            {
+                case "Trace":
+                    ret = LogLevel.Trace;
+                    break;
+                case "Debug":
+                    ret = LogLevel.Debug;
+                    break;
+                case "Information":
+                    ret = LogLevel.Information;
+                    break;
+                case "Warning":
+                    ret = LogLevel.Warning;
+                    break;
+                case "Error":
+                    ret = LogLevel.Error;
+                    break;
+                case "Critical":
+                    ret = LogLevel.Critical;
+                    break;
+                default:
+                    if (!int.TryParse(levelString, out level))
+                    {
+                        return false;
+                    }
+                    if (!(LogLevel.Trace <= (LogLevel)level && (LogLevel)level <= LogLevel.None))
                     {
-                        cur.SetFilterSpec(_filterSpec);
+                        return false;
                     }
-                    _checkLevel = false;
+                    ret = (LogLevel)level;
+                    break;
+            }
+            return true;
+        }
+
+        [NonEvent]
+        private LogLevel GetDefaultLevel()
+        {
+            var allMessageKeywords = Keywords.Message | Keywords.FormattedMessage | Keywords.JsonMessage;
+
+            if (IsEnabled(EventLevel.Informational, allMessageKeywords))
+            {
+                if (IsEnabled(EventLevel.Verbose, allMessageKeywords))
```

Comment:
Is this funky branching just to avoid checking Verbose unless we know we're at least enabled for Informational?

Reply:
Copied from old implementation. I'll flattened it all the way and add mapping from LogAlways to Trace.